waku/v2/protocol/lightpush/waku_lightpush_test.go

```diff
@@ -332,3 +314,64 @@ func TestWakuLightPushCornerCases(t *testing.T) {
 	// Test situation when cancel func is nil
 	lightPushNode2.cancel = nil
 }
+
+func TestWakuLightPushWithStaticSharding(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Prepare pubsub topics for static sharding
+	pubSubTopic := protocol.NewStaticShardingPubsubTopic(uint16(25), uint16(0)).String()
+	testContentTopic := "/test/10/my-lp-app/proto"
+
+	node1, sub1, host1 := makeWakuRelay(t, pubSubTopic)
+	defer node1.Stop()
+	defer sub1.Unsubscribe()
+
+	node2, sub2, host2 := makeWakuRelay(t, pubSubTopic)
```

Comment:
Node2 doesn't need to be a relay node if it wants to use lightpush for publishing. 
It can just be a simple lightpush node and use node1 to publish.

Reply:
Node2 is a forwarder to node1, which is only relay.  I've tried to change node2 to be only lightpush, and received confusing error. https://github.com/waku-org/go-waku/pull/1060/commits/73a90e085c8b90418abab36e2abea17abee74d58

```
--- FAIL: TestWakuLightPushWithStaticSharding (2.47s)
    waku_lightpush_test.go:368:
        	Error Trace:	/Users/roman/sources/waku-org/go-waku/waku/v2/protocol/lightpush/waku_lightpush_test.go:368
        	Error:      	Received unexpected error:
        	            	failed to negotiate protocol: protocols not supported: [/vac/waku/lightpush/2.0.0-beta1]
        	Test:       	TestWakuLightPushWithStaticSharding
```

Have I missed something ?

Reply:
Ah..ok. 
If so, then node2 has to be support relay and lightpush as service node.

Reply:
Reorganized at https://github.com/waku-org/go-waku/pull/1060/commits/093ea363a6ac347c0c424360a34075bfed62a4a7